.github/workflows/auto-set-base.yml

```diff
@@ -23,7 +23,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
         run: |
-          if [ "$TARGET_BRANCH" = "main" ] && [ "$BASE_BRANCH" != "develop" ]; then
+          if [[ "$TARGET_BRANCH" == "main" && "$BASE_BRANCH" != "develop" && ! "$BASE_BRANCH" == rc-* ]]; then
```

Comment:
Is this `!` before the expression common for bash? rather than `!=` which we're using in the develop conditional?
If we can use `!=` instead I think we should for consistency.

Reply:
good question! we have to use `!` before `"$BASE_BRANCH" == rc-*` since that's what used for pattern matching. In other conditionals we have here, like checking `main` and `develop` branch we can use the `==` and `!=` directly

Reply:
I ran some tests locally and it seems `!=` will work fine for pattern matching too

Reply:
https://www.namehero.com/blog/bash-string-comparison-the-comprehensive-guide/#3-3-partial-string-matching

Reply:
I think I was missing the `""` around `rc-*` when I was trying out the `!=.` it seems to work now with the quotes. thanks Mike!